### ISSUE TITLE
Document native histogram features

### ISSUE THREAD

**Issue by beorn7** [prometheus/prometheus#11173]
Document the feature flag itself, and the various new features. Some is already done, some is marked with TODOs, some hasn’t been touched at all.

**Comment by beorn7** [comment]
I'm now doing a sweep through the codebase to find remaining documentation needs. I'll either add documentation directly or add TODO items to this issue.

**Comment by beorn7** [comment]
Relevant PRs so far:

- https://github.com/prometheus/prometheus/pull/11452
- https://github.com/prometheus/prometheus/pull/11446

Still TODO:
- [ ] Mention that federation doesn't support native histograms yet.
- [ ] Mention that native histograms cannot be the output of a recording rule.
- [ ] Document the JSON format of native histograms in the query API.
- [ ] Pass through the general documentation in prometheus/docs and mention native histograms where appropriate.

**Comment by beorn7** [comment]
- https://github.com/prometheus/prometheus/pull/11454
- https://github.com/prometheus/prometheus/pull/11453

**Comment by beorn7** [comment]
And finally https://github.com/prometheus/docs/pull/2218 .

I think this should be enough for the merge to main.

**Comment by beorn7** [comment]
https://github.com/prometheus/docs/pull/2218 is not merged yet, but only because we want to wait on the merge in prometheus/prometheus to avoid confusion.

Closing this now to unblock the merge of https://github.com/prometheus/prometheus/pull/11447
